sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -0,0 +1,181 @@
+
+        let authPolicy = CommunicationUserCredentialPolicy(credential: credential)
+
+        self.client = try AzureCommunicationChatClient(baseUrl: baseUrl, authPolicy: authPolicy, withOptions: options)
+    }
+
+    // MARK: Public Methods
+
+    /// Get a ChatThreadClient.
+    /// - Parameters:
+    ///   - threadId: The threadId.
+    public func getChatThreadClient(threadId: String) throws -> ChatThreadClient {
+        return try ChatThreadClient(
+            threadId: threadId,
+            endpoint: endpoint,
+            credential: credential,
+            withOptions: options
+        )
+    }
+
+    /// Create a ChatThread.
+    /// - Parameters:
+    ///   - request: Request for creating a chat thread with the topic and members to add.
+    ///   - withOptions: Create chat thread options.
+    ///   - completionHandler: A completion handler that receives a ChatThreadClient on success.
+    public func createChatThread(
+        request: CreateChatThreadRequest,
+        withOptions options: CreateChatThreadOptions? = nil,
+        completionHandler: @escaping (Result<ChatThreadClient, Error>) -> Void
+    ) {
+        client.create(chatThread: request, withOptions: options) { result, _ in
+            do {
+                switch result {
+                case let .success(createThreadResponse):
+                    var responseId: String?
+                    for response in createThreadResponse.multipleStatus ?? [] {
+                        if response.id?.hasSuffix("@thread.v2") ?? false,
+                           response.type ?? "" == "Thread" {
+                            responseId = response.id
+                        }
+                    }
+
+                    guard let threadId = responseId else {
+                        throw AzureError.service("Service response does not contain ThreadId.")
+                    }
+
+                    let chatThreadClient = try ChatThreadClient(
+                        threadId: threadId,
+                        endpoint: self.endpoint,
+                        credential: self.credential,
+                        withOptions: self.options
+                    )
+                    completionHandler(.success(chatThreadClient))
+
+                case let .failure(error):
+                    throw error
+                }
+            } catch {
+                completionHandler(.failure(error))
+            }
+        }
+    }
+
+    /// Get a ChatThread.
+    /// - Parameters:
+    ///   - chatThreadId: The chat thread id.
+    ///   - withOptions: Get chat thread options.
+    ///   - completionHandler: A completion handler that receives the chat thread on success.
+    public func getChatThread(
+        chatThreadId: String,
+        withOptions options: GetChatThreadOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<ChatThread>
+    ) {
+        client.getChatThread(chatThreadId: chatThreadId, withOptions: options) { result, httpResponse in
```

Comment:
The generated code here doesn't produce the right signature. I would recommend either of the following:
```swift
public func getChatThread(withId: String, withOptions options: ...)
public func get(chatThreadId: String, withOptions options: ...)
```

@bsiegel do you have a thought on this? I don't think our guidelines match this precisely.